pyquil/tests/test_paulis.py

```diff
@@ -330,6 +331,13 @@ def test_exponentiate():
     assert prog == result_prog
 
 
+def test_exponentiate_pauli_sum():
+    pauli_sum = PauliSum([PauliTerm('X', 0, 0.5), PauliTerm('Z', 0, 0.5)])
```

Comment:
This unit test uses non-commuting Pauli-terms, so please modify according to your decision regarding the name of this function and how to document it.